### PR TITLE
Truthsort with arrays

### PR DESCRIPTION
**Pull request by jhowl01**
Currently, the output of `TruthSorting.py` is a minitree with, for instance, separate columns for "largest_s1_area" and "largest_s2_area". This branch instead outputs a minitree with parallel array columns for peak-level data, like `area = [67, 421]` and `type = [1, 2]` (type = 1 for s1, and 2 for s2). While this sacrifices some of pandas's straightforward cut logic for non-array columns, it keeps a row-per-event structure that can be easily merged with minitrees from processed data, especially data with more than two peaks per event.

Usage has not changed - if the .root output format is selected, writing to file is now done using `hax.minitree_formats.dataframe_to_root()`.

### CODE REVIEW

File: montecarlo/fax_waveform/TruthSorting.py [hunk 1]
```diff
@@ -61,18 +63,12 @@
...
-Data['s2_electron_time_truth'] = [] 
-Data['s2_first_electron_time_truth'] = [] 
-Data['s2_time_std_truth'] = [] 
-Data['s2_area_truth'] = [] 
-Data['s2_bottom_area_truth'] = []
-Data['x_truth'] = []
-Data['y_truth'] = []
+event_keys = ['index_truth']
+s1s2_keys = ['time_truth', 'time_std_truth', 'area_truth', 'type_truth']
+s2_only_keys = ['electron_time_truth', 'first_electron_time_truth', 'bottom_area_truth', 'x_truth', 'y_truth']
```
Comment (pdeperio) [inline montecarlo/fax_waveform/TruthSorting.py]: can you please add 'z_truth' (and any other variables that may be useful)?

**Comment by jhowl01** [comment]
Things I'm not sure about:
- Fields from the fax_truth.root files we don't currently save: t_interaction, t_last_electron, t_last_photon
- Currently both methods only save S2 position information. S1's have truth (x, y, z) too, although if you use Qing's S1/S2 correlation they are the same. Should we save these as well?
- I opted for having fields like `area=[first_area, second_area, ...]`, `width=[first_width, second_width, ...]`, `type=[1, 2, ...]`, instead of `s1_area=[first_s1_area, second_s1_area, ...]`, `s2_area=[first_s2_area, second_s2_area, ...]`.

**Comment by pdeperio** [comment]
- Save these too please. Imagine plotting a (averaged) waveform and overlaying lines for these true times
- Save these as well. Consider multiple scatters at different positions (and we're not yet creating any "interaction" object to match S1's and S2's).
- I think this is better, considering the possibility of no matching S2 to an S1 (interaction outside of drift field). Related to this then, can you save just a single length variable like "peaks_length" that defines the size of all branch arrays?

**Comment by jhowl01** [comment]
Made suggested changes. Note that root output still adds extra "length" branches despite the "peaks_length" column.

**Comment by pdeperio** [comment]
so no way of disabling writing out the extra length branches in hax?

**Comment by jhowl01** [comment]
In the conversation [here](https://github.com/XENON1T/hax/pull/32), we landed on this extra branches method as the best way of writing arrays to a .root. There was no robust check we could use to find the correct length-branch when the user supplied it. The only workaround I can think of is to copy the function over from hax into `TruthSorting_arrays.py` and change it to disable this.

**Comment by pdeperio** [comment]
Fixes #6

**Pull request merged**